### ISSUE TITLE
[feature] List / delete applications

### ISSUE THREAD

**Issue by drym3r** [superseriousbusiness/gotosocial#2523]
### Is your feature request related to a problem ?

I'd like to list all the apps that I've created and even delete them.

### Describe the solution you'd like.

I'd like more endpoints under `/api/v1/apps` that let list, describe, edit and delete any app you may have.

### Describe alternatives you've considered.

Not sure there's any. 

### Additional context.

My assumption is that's there's no way to do what I say after reading the OAS, which only shows the `/api/v1/apps/` endpoint accepting the POST verb. If that's not the case, sorry for the noise and please point me in the right direction.

**Comment by tsmethurst** [comment]
I've added this to the OAuth improvements issue.

**Comment by jcarnat** [comment]
Hi,

AFAIK this is not available in the APIs yet.

In the meantime, what is the proper way to list and delete applications?
Shall one simply browse the `applications` table to get the list of authorized apps and delete records from it to remove authorization?

(I'm using sqlite if that matters)

**Comment by alimoeeny** [comment]
is there any potential issues with deleting a row from the applications table?

**Comment by tsmethurst** [comment]
> is there any potential issues with deleting a row from the applications table?

Yes, it will probably cause issues with tokens created by that application. You can do it, but delete the tokens the application created as well.
